src/movegen/rooks.h

```diff
@@ -92,27 +99,65 @@ constexpr auto generateRookAttackTable()
 
         for (int index = 0; index < occupancyIndicies; index++) {
             const uint64_t occupancy = magic::set_occupancy(index, relevantBitsCount, attackMask);
-            const int magicIndex = (occupancy * magic::s_rooksMagic[square]) >> (64 - magic::s_rookRelevantBits[square]);
 
+            // pext_u64 is not available at compile time
+            int magicIndex = 0, bit = 1;
+            for (uint64_t m = attackMask; m; m &= (m - 1)) {
+                if (occupancy & (m & -m)) {
+                    magicIndex |= bit;
+                }
+                bit <<= 1;
+            }
             attacks[square][magicIndex] = rookAttacksWithBlock(square, occupancy);
         }
     }
 
     return attacks;
 }
+#else
+constexpr std::array<magic::SliderAttackTable, 64> generateRookAttackTable()
+{
+    std::array<magic::SliderAttackTable, 64> attacks;
 
-constexpr auto s_rookAttackTable = generateRookAttackTable();
+    for (int square = 0; square < 64; square++) {
 
+        const uint64_t attackMask = s_rookMasksTable[square];
+        const int relevantBitsCount = std::popcount(attackMask);
+        const int occupancyIndicies = (1 << relevantBitsCount);
+
+        for (int index = 0; index < occupancyIndicies; index++) {
+            const uint64_t occupancy = magic::set_occupancy(index, relevantBitsCount, attackMask);
+            const int magicIndex = (occupancy * magic::hashing::s_rooksMagic[square]) >> (64 - magic::hashing::s_rookRelevantBits[square]);
+
+            attacks[square][magicIndex] = rookAttacksWithBlock(square, occupancy);
+        }
+    }
+
+    return attacks;
 }
+#endif
+
+constexpr auto s_rookAttackTable = generateRookAttackTable();
+} // end namespace
 
+#ifdef USE_BMI2
+// https://www.chessprogramming.org/BMI2#PEXT_Bitboards
+static inline uint64_t getRookMoves(BoardPosition pos, uint64_t occupancy)
+{
+    occupancy = _pext_u64(occupancy, s_rookMasksTable[pos]);
+
+    return s_rookAttackTable[pos][occupancy];
+}
+#else
 /* https://www.chessprogramming.org/Magic_Bitboards */
 static inline uint64_t getRookMoves(BoardPosition pos, uint64_t occupancy)
```

Comment:
These methods have the same signature, so I thought we might as well move them to their respective headers and simply use "rooks.h" to include the methods that we're currently compiling for?
Then there'll basically be nothing in this file but merely the includes meaning that it will be completely agnostic from the consumers of the interface :)